### PR TITLE
test(cli): clean up unit and e2e for CLI commands

### PR DESCRIPTION
**Pull request by Tlacenka**
Related to #211 

This PR covers only part of the test refactor. Creating a shared setup, unifying mocks and cleaning up the rest of the CLI package tests is yet to come.

In this PR I did the following:
- Reduced purpose of unit tests only to the relevant CLI command.
- Reduced boiler-plate code in tests.
- Used `fs` and console mocks to test the command behaviour in unit tests.
- Reduced the number of E2E tests for `print-config` (covered already in unit tests).
- Fixed `filterKebabCaseKeys` and updated its unit tests.

### CODE REVIEW

File: packages/cli/src/lib/autorun/autorun-command.unit.test.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+});
+vi.mock('fs/promises', async () => {
+  const memfs: typeof import('memfs') = await vi.importActual('memfs');
+  return memfs.fs.promises;
+});
+
+vi.mock('@code-pushup/core', async () => {
+  const core = await vi.importActual('@code-pushup/core');
+  return {
+    ...(core as object),
```
Comment (matejchalk) [inline packages/cli/src/lib/autorun/autorun-command.unit.test.ts:41]: Tiny thing, but we can easily make the type more specific.

```suggestion
  const core: typeof import('@code-pushup/core') =
    await vi.importActual('@code-pushup/core');
  return {
    ...core,
```

File: packages/cli/src/lib/autorun/autorun-command.unit.test.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+});
+
+describe('autorun-command', () => {
+  beforeEach(async () => {
+    vi.clearAllMocks();
+    vol.reset();
+    vol.fromJSON(
+      {
+        'my-report.json': JSON.stringify(report()),
+        'code-pushup.config.ts': '', // only needs to exist for stat inside bundleRequire
```
Comment (matejchalk) [inline packages/cli/src/lib/autorun/autorun-command.unit.test.ts:64]:
```suggestion
        'code-pushup.config.ts': '', // only needs to exist for stat inside readCodePushupConfig
```

File: packages/cli/src/lib/autorun/autorun-command.unit.test.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    await vi.importActual('@code-pushup/models/testing');
+  return {
+    bundleRequire: vi
+      .fn()
+      .mockResolvedValue({ mod: { default: minimalConfig() } }),
+  };
+});
+
+describe('autorun-command', () => {
+  beforeEach(async () => {
```
Comment (matejchalk) [inline packages/cli/src/lib/autorun/autorun-command.unit.test.ts:58]: There's no `await`, so no need for `async`. (We have a lint rule for this, but errors need fixing first.)

```suggestion
  beforeEach(() => {
```

File: packages/cli/src/lib/yargs-cli.integration.test.ts [hunk 4]
```diff
@@ -1,69 +1,104 @@
...
-    const args: string[] = ['--config=./config.a.ts'];
-    const parsedArgv = await yargsCli(args, {
-      options,
-    }).parseAsync();
+  it('should parse a single config argument as a string', async () => {
+    const parsedArgv = await yargsCli<GeneralCliOptions>(
+      ['--config=./config.a.ts'],
+      {
+        options,
+      },
```
Comment (matejchalk) [inline packages/cli/src/lib/yargs-cli.integration.test.ts:31]: Tiny formatting improvement, saves some real estate.

The trick is to put the 1st property on the same line as `{` before you format, then Prettier will try to put it on one line. But if the 1st property is already on another line, Prettier always formats it as multiline.

```suggestion
      { options },
```

**Review by matejchalk — DISMISSED** [review]
I :heart: this refactor :tada:

File: packages/cli/src/lib/autorun/autorun-command.unit.test.ts (hunk 1 above)
Comment (Tlacenka) [reply packages/cli/src/lib/autorun/autorun-command.unit.test.ts:41]: When I tried to do that, I got the `Static imports of lazy-loaded libraries are forbidden` lint error :(

File: packages/cli/src/lib/autorun/autorun-command.unit.test.ts (hunk 2 above)
Comment (Tlacenka) [reply packages/cli/src/lib/autorun/autorun-command.unit.test.ts]: Good point, fixed!

File: packages/cli/src/lib/autorun/autorun-command.unit.test.ts (hunk 3 above)
Comment (Tlacenka) [reply packages/cli/src/lib/autorun/autorun-command.unit.test.ts]: Nice catch!

File: packages/cli/src/lib/yargs-cli.integration.test.ts (hunk 4 above)
Comment (Tlacenka) [reply packages/cli/src/lib/yargs-cli.integration.test.ts]: I didn't know about this, that's a game changer 😮

File: packages/cli/src/lib/autorun/autorun-command.unit.test.ts (hunk 1 above)
Comment (matejchalk) [reply packages/cli/src/lib/autorun/autorun-command.unit.test.ts:41]: Oh, interesting :D OK, then leave it as is, it doesn't matter very much in this case.

**Review by matejchalk — APPROVED** [review]

**Pull request merged**